be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala

```diff
@@ -2,28 +2,82 @@ package ch.epfl.pop.pubsub.graph
 
 import akka.NotUsed
 import akka.stream.scaladsl.Flow
+import ch.epfl.pop.model.network.method.{Broadcast, Catchup, Publish, Subscribe, Unsubscribe}
+import ch.epfl.pop.model.network.{JsonRpcRequest, JsonRpcResponse}
 import ch.epfl.pop.model.network.requests.lao.{JsonRpcRequestCreateLao, JsonRpcRequestStateLao, JsonRpcRequestUpdateLao}
 import ch.epfl.pop.model.network.requests.meeting.{JsonRpcRequestCreateMeeting, JsonRpcRequestStateMeeting}
 import ch.epfl.pop.model.network.requests.witness.JsonRpcRequestWitnessMessage
-import ch.epfl.pop.pubsub.graph.validators.LaoValidator.{validateCreateLao, validateStateLao, validateUpdateLao}
-import ch.epfl.pop.pubsub.graph.validators.MeetingValidator.{validateCreateMeeting, validateStateMeeting}
-import ch.epfl.pop.pubsub.graph.validators.RollCallValidator.{validateCloseRollCall, validateCreateRollCall, validateOpenRollCall, validateReopenRollCall}
-import ch.epfl.pop.pubsub.graph.validators.WitnessValidator.validateWitnessMessage
+
+import ch.epfl.pop.pubsub.graph.validators.RpcValidator._
+import ch.epfl.pop.pubsub.graph.validators.MethodValidator._
+import ch.epfl.pop.pubsub.graph.validators.MessageValidator._
+import ch.epfl.pop.pubsub.graph.validators.LaoValidator._
+import ch.epfl.pop.pubsub.graph.validators.MeetingValidator._
+import ch.epfl.pop.pubsub.graph.validators.RollCallValidator._
+import ch.epfl.pop.pubsub.graph.validators.WitnessValidator._
 
 object Validator {
 
+  private val VALIDATOR_ERROR: PipelineError = PipelineError(
+    ErrorCodes.INVALID_ACTION.id,
+    "Internal server fault: Validator was given a message it could not recognize"
```

Comment:
`Internal server fault` might not be the best description here, how about `Unsupported action` ?